manage_breast_screening/mammograms/forms/implanted_medical_device_history_form.py

```diff
@@ -0,0 +1,139 @@
+import datetime
+
+from django import forms
+from django.db.models import TextChoices
+from django.forms.widgets import Textarea
+
+from manage_breast_screening.core.services.auditor import Auditor
+from manage_breast_screening.nhsuk_forms.fields import (
+    CharField,
+    ChoiceField,
+    IntegerField,
+)
+from manage_breast_screening.nhsuk_forms.fields.choice_fields import (
+    MultipleChoiceField,
+)
+from manage_breast_screening.nhsuk_forms.forms import FormWithConditionalFields
+from manage_breast_screening.participants.models.implanted_medical_device_history_item import (
+    ImplantedMedicalDeviceHistoryItem,
+)
+
+
+class RemovalStatusChoices(TextChoices):
+    HAS_BEEN_REMOVED = "HAS_BEEN_REMOVED", "Implanted device has been removed"
+
+
+class ImplantedMedicalDeviceHistoryForm(FormWithConditionalFields):
+    def __init__(self, *args, participant, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        # if entered, years should be between 80 years ago and this year
```

Comment:
What do you think about extracting some of this logic so we can reuse it in the other forms? E.g. we could perhaps have some functions for producing those out of range error messages. Generally all the error messages will follow a similar format.

The magic number of 80 could be moved to a constant somewhere as well.